### ISSUE TITLE
Unprocessable Entity (422) when uploading file with non-English name

### ISSUE THREAD

**Issue by tenniscp25** [http4s/http4s#1767]
With the following code:

```scala
import cats.effect._
import org.http4s._
import org.http4s.dsl.io._
import org.http4s.multipart._

object TestService {
  val service = HttpService[IO] {
    case req @ POST -> Root / "test" =>
      req.decode[Multipart[IO]](_ => Ok("Done"))
  }
}
```

I tried Thai, Japanese and Chinese file names. The server always return HTTP 422. English names work fine though. Example names that fail: ทดสอบ, テスト and 实验.

**Comment by jmcardon** [comment]
Hi.

The [Multipart RFC](https://tools.ietf.org/html/rfc7578#section-5.1.1) suggests to use ascii names when possible, and if not using ascii, at least being consistent.

Currently, we only support ascii, but I'm opening a PR in a little to tackle that issue

**Comment by rossabaker** [comment]
We're not bumping into 5.1.3 of that spec yet, are we?  I think we've got some unresolved issues around that, as well.

**Comment by jmcardon** [comment]
@tenniscp25 #1768 addresses your issue.

**Comment by jmcardon** [comment]
Closed by #1769. Please upgrade to 0.18.7 to deal with non-ascii chars :+1:

**Comment by tenniscp25** [comment]
Thanks!
